R/scATAC.Peak.annotation.R

```diff
@@ -369,36 +375,43 @@ peaks_on_gene <- function(peak_features,annotations=NULL, gene_element=NULL, spl
           vorzeichen[which(vorzeichen == "")] <- "." 
           distances2 <- lapply(distances2, function(x) {paste(x, collapse = "|")})
           distances2 <- unlist(distances2)
-          gene_mid1 <- paste0(gene_mid, "_",chr_index[[chr]][["strand"]][index],"_","TSS.dist.Pstart.","_",as.character(distances1), "_",vorzeichen)
-          gene_mid2 <- paste0(gene_mid, "_", chr_index[[chr]][["strand"]][index],"_","TSS.dist.Pend.","_",as.character(distances2), "_",vorzeichen)
+          gene_mid1 <- paste0(gene_mid, ";", gene_names, ";",chr_index[[chr]][["strand"]][index], ";", "TSS.dist.Pstart.", ";", as.character(distances1), ";", vorzeichen)
+          gene_mid2 <- paste0(gene_mid, ";", gene_names, ";", chr_index[[chr]][["strand"]][index], ";", "TSS.dist.Pend.", ";", as.character(distances2), ";", vorzeichen)
           gene_mid <- c(gene_mid1, gene_mid2)
         }
-        gene_names <- c(gene_left, gene_right, gene_mid)
-        if (length(gene_names) > 1) { gene_names <- unique(gene_names)}
-        if (length(gene_names) == 1) {
-          return(c(peak[x,],unlist(strsplit(gene_names, split = "_"))))
+        gene_ids <- c(gene_left, gene_right, gene_mid)
+        if (length(gene_ids) > 1) { gene_ids <- unique(gene_ids)}
+        if (length(gene_ids) == 1) {
+          return(c(peak[x,],unlist(strsplit(gene_ids, split = ";"))))
         }
-        else if (length(gene_names) == 0) {
-          return(c(peak[x,], "nomatch", ".", ".", ".","."))
+        else if (length(gene_ids) == 0) {
+          return(c(peak[x,], "nomatch", "nomatch", ".", ".", ".","."))
         }
         else {
-          to_return <- matrix(rep(peak[x,],length(gene_names)),nrow = length(gene_names), byrow = T)
-          to_return <- cbind(to_return, t(sapply(gene_names, function(x) { y <- strsplit(x, split="_"); y <- unlist(y) })))
+          to_return <- matrix(rep(peak[x,], length(gene_ids)), nrow=length(gene_ids), byrow=T)
+          to_return <- cbind(to_return, t(sapply(gene_ids, function(x) { y <- strsplit(x, split=";"); y <- unlist(y) })))
           return(to_return)
         }
       }
       else{
-        gene_names <- c(gene_left, gene_right, gene_mid)
-        if (length(gene_names) > 1) { gene_names <- unique(gene_names)}
-        if (length(gene_names) == 1) {
-          return(c(peak[x,],gene_names))
+        gene_ids <- c(gene_left, gene_right, gene_mid)
+        gene_names <- chr_index[[chr]][["gene_names"]][(left1 & left2) & !(mid1 & mid2)]
+        gene_names <- c(gene_names, chr_index[[chr]][["gene_names"]][(right1 & right2) & !(mid1 & mid2)])
```

Comment:
repetitive booleans removed as initial conditions are changed to non-redundant

Reply:
Done